controllers/storagecluster/storagecluster_controller.go

```diff
@@ -166,13 +166,17 @@ func (r *StorageClusterReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		},
 	}
 
-	onboardingSecretPredicates := builder.WithPredicates(
-		predicate.NewPredicateFuncs(
-			func(client client.Object) bool {
-				return client.GetName() == onboardingTicketPublicKeySecretName
-			},
-		),
-	)
+	secretPredicates := builder.WithPredicates(predicate.Funcs{
```

Comment:
why to use `builder.WithPredicates` here itself? better to follow exisitng pattern, this here locks the predicate to be only one iiuc. not mandatory if you aren't going to refresh.

Reply:
I have added this because ohad said this here, https://github.com/red-hat-storage/ocs-operator/pull/2385#discussion_r1466285752

Reply:
@rchikatw I was wrong. the `WithPredicates` takes a list of `predicate.Funcs` structure instances with that new knowledge in mind I agree with @leelavg